File: 1557-check-if-a-string-contains-all-binary-codes-of-size-k/check-if-a-string-contains-all-binary-codes-of-size-k.py
class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        subs_count = 2 ** k
        seen = set()
        for i in range(len(s) - k + 1):
            seen.add(s[i: i+k])
        return len(seen) == subs_count

File: 1557-check-if-a-string-contains-all-binary-codes-of-size-k/check-if-a-string-contains-all-binary-codes-of-size-k.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
        if (s.size() < k)
            return false;
        std::unordered_set<std::string> seen;
        for (size_t i = 0; i < s.size() - k + 1; ++i) {
            seen.insert(s.substr(i, k));
        }
        return seen.size() == (unsigned int)std::pow(2, k);
    }
};